### PR TITLE
First try to fix linter errors

### PR DESCRIPTION
**Pull request by AugustasV**
#### Summary
Uploaded Golang lint build step to the project and tried to fix some of linter errors. Would like to hear suggestions.

#### Ticket Link

  First try https://github.com/mattermost/mattermost-plugin-google-calendar/issues/13

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,9 +4,20 @@ go 1.12
 
 require (
 	github.com/google/uuid v1.1.1
+	github.com/gorilla/websocket v1.4.2 // indirect
```
Comment (hanzei) [inline go.mod:7]: Why are all of these indirect dependencies getting added?

File: server/api.go [hunk 2]
```diff
@@ -114,28 +114,31 @@ func (p *Plugin) completeCalendar(w http.ResponseWriter, r *http.Request) {
...
-	p.startCronJob(authedUserId)
+	p.startCronJob(autheduserID)
 
 	// Post intro post
 	message := "#### Welcome to the Mattermost Google Calendar Plugin!\n" +
 		"You've successfully connected your Mattermost account to your Google Calendar.\n" +
 		"Please type **/calendar help** to understand how to user this plugin. "
 
-	p.CreateBotDMPost(userId, message)
+	if err = p.CreateBotDMPost(userID, message); err != nil {
```
Comment (hanzei) [inline server/api.go:138]: I think it makes sense to remove the returned error from `p.CreateBotDMPost` and let the method itself log it.

File: server/api.go [hunk 3]
```diff
@@ -191,10 +194,7 @@ func (p *Plugin) handleEventResponse(w http.ResponseWriter, r *http.Request) {
 	srv, _ := p.getCalendarService(userID)
 
 	eventToBeUpdated, err := srv.Events.Get(calendarID, eventID).Do()
-	if err != nil {
```
Comment (hanzei) [inline server/api.go:194]: Why this change?

File: server/api.go [hunk 4]
```diff
@@ -214,16 +214,18 @@ func (p *Plugin) handleEventResponse(w http.ResponseWriter, r *http.Request) {
 }
 
 func (p *Plugin) watchCalendar(w http.ResponseWriter, r *http.Request) {
-	userID := r.URL.Query().Get("userId")
+	userID := r.URL.Query().Get("userID")
```
Comment (hanzei) [inline server/api.go:217]: I don't think that is something we can just change. The query parameter is set from google.

File: server/api.go (hunk 3 above)
Comment (AugustasV) [reply server/api.go:194]: I tought it would be good idea to insert if error command, just in case later someone would like to write function for it.

File: go.mod (hunk 1 above)
Comment (AugustasV) [reply go.mod:7]: I think it could be `go mod tidy` command fault.

Comment (hanzei) [reply go.mod:7]: Would you please remove them.

File: server/api.go (hunk 4 above)
Comment (AugustasV) [reply server/api.go:217]: Ok, I just saw linter error, that's why I tought it's good idea to change it
```
server/api.go:82:2: var `userId` should be `userID` (golint)
        userId := strings.Split(state, "_")[1]
```

**Comment by AugustasV** [comment]
@hanzei thanks for comments, any direction how I should fix remaining linter warnings?

**Comment by hanzei** [comment]
@AugustasV You can comment out the `goconst` linter as it's not high value and 	hard to fix sometimes. You can also disable `errcheck` as it take a lot of time to fixe those.

The rest should be easy to fix. `gosimple` and `golint` make up for the most of the remaining issues. 
Let me know if you need any additional help.

**Comment by AugustasV** [comment]
@hanzei disabled linters which was hard to fix. Now linter is passing.

File: go.mod [hunk 5]
```diff
@@ -4,9 +4,19 @@ go 1.12
 
 require (
 	github.com/google/uuid v1.1.1
+	github.com/klauspost/compress v1.10.5 // indirect
```
Comment (hanzei) [inline go.mod:7]: Could you please remove the `indirect` dependencies?

File: server/api.go (hunk 3 above)
Comment (hanzei) [reply server/api.go:194]: I still don't quite get the change. Shouldn't this message only be printed in an error case?
